static/assets/js/entity-reference/app.js

```diff
@@ -0,0 +1,82 @@
+(function(){
+    "use strict";
+
+    var app = angular.module('entityReferenceApp', []);
+
+    app.controller('EntityReferenceCtrl', ['$http', EntityReferenceCtrl]);
+
+    app.component('bbEntityReference', {
+        templateUrl: '/assets/views/entities.html',
+        controller: 'EntityReferenceCtrl as ctrl',
+        bindings: {
+            apiTitle: '@',
+            lastUpdatedDate: '@',
+            getStartedUrl: '@',
+            swaggerUrl: '@',
+            whiteList: '@',
+            blackList: '@'
+        }
+    });
+
+    function EntityReferenceCtrl($http) {
+        this.title = 'Entity Reference';
+        this.showErrorMessage = false;
+
+        $http.get(this.swaggerUrl).then(handleSuccess.bind(this), handleError.bind(this));
+
+        function handleSuccess(response) {
+            var swagger = response.data;
+            this.swagger = swagger;
+            var whiteList = this.whiteList ? this.whiteList.split(',') : [];
+            var blackList = this.blackList ? this.blackList.split(',') : [];
+            this.entities = getEntitiesFromSwagger(swagger, whiteList, blackList);
+        }
+
+        function handleError(response) {
+            this.showErrorMessage = true;
+        }
+
+        function getEntitiesFromSwagger(swagger, whiteList, blackList) {
+            return Object.keys(swagger.definitions)
+                .sort()
+                .map(function(name) {
+                    return { name: name, details: swagger.definitions[name] };
+                })
+                .filter(function(entity) {
+                    if (whiteList.length > 0) {
+                        return whiteList.find(function(name) {
+                            return entity.name == name;
+                        });
+                    }
+                    if (blackList.length > 0) {
+                        return !blackList.find(function(name) {
+                            return entity.name == name;
+                        });
+                    }
+                    return entity;
+                })
+                .map(setDisplayTypesOnEntity);
+        }
+
+        function setDisplayTypesOnEntity(entity) {
+            Object.keys(entity.details.properties).forEach(function(propertyName) {
+                var property = entity.details.properties[propertyName];
+
+                if (property.type == "array") {
+                    var ref = property.items.$ref.replace("#/definitions/", "");
+                    property.displayType = pluralize(ref, 2);
```

Comment:
So, we actually have the link text singular intentionally in order to match the wording in the entity header it links to. 
This also helps clear up confusion between the `User` and `Users` entities.
Things like strings and ints are also left singular for consistency. So this should not be pluralized.